contracts/script/Manage.s.sol

```diff
@@ -0,0 +1,496 @@
+///     * EXECUTOR address of executor
+///     * ADMIN address of admin (optional; leave unset to disable)
+///
+/// See the Foundry documentation for more information about Solidity scripts.
+/// https://book.getfoundry.sh/tutorials/solidity-scripting
+contract DeployTimelockRouter is Script {
+    function run() external {
+        vm.startBroadcast();
+
+        // initial minimum delay in seconds for operations
+        uint256 minDelay = vm.envUint("MIN_DELAY");
+        console2.log("minDelay:", minDelay);
+
+        // accounts to be granted proposer and canceller roles
+        address[] memory proposers = new address[](1);
+        proposers[0] = vm.envAddress("PROPOSER");
+        console2.log("proposers:", proposers[0]);
+
+        // accounts to be granted executor role
+        address[] memory executors = new address[](1);
+        executors[0] = vm.envAddress("EXECUTOR");
+        console2.log("executors:", executors[0]);
+
+        // optional account to be granted admin role; disable with zero address
+        address admin = vm.envOr("ADMIN", address(0));
+        console2.log("admin:", admin);
+
+        // Deploy new contracts
+        TimelockController timelockController = new TimelockController(minDelay, proposers, executors, admin);
+        console2.log("Deployed TimelockController to", address(timelockController));
+
+        RiscZeroVerifierRouter verifierRouter = new RiscZeroVerifierRouter(address(timelockController));
+        console2.log("Deployed RiscZeroVerifierRouter to", address(verifierRouter));
+
+        vm.stopBroadcast();
+    }
+}
+
+/// @notice Deployment script for the RISC Zero verifier with Emergency Stop mechanism.
+/// @dev Use the following environment variable to control the deployment:
+///     * SCHEDULE_DELAY (optional) minimum delay in seconds for the scheduled action
+///     * VERIFIER_ESTOP_OWNER owner of the emergency stop contract
+///     * TIMELOCK_CONTROLLER contract address of TimelockController
+///     * VERIFIER_ROUTER contract address of RiscZeroVerifierRouter
+///
+/// See the Foundry documentation for more information about Solidity scripts.
+/// https://book.getfoundry.sh/tutorials/solidity-scripting
+contract DeployEstopVerifier is Script {
+    function run() external {
+        vm.startBroadcast();
+
+        address verifierEstopOwner = vm.envAddress("VERIFIER_ESTOP_OWNER");
+        console2.log("verifierEstopOwner:", verifierEstopOwner);
+
+        // Locate contracts
+        TimelockController timelockController = TimelockController(payable(vm.envAddress("TIMELOCK_CONTROLLER")));
+        console2.log("Using TimelockController at address", address(timelockController));
+
+        RiscZeroVerifierRouter verifierRouter = RiscZeroVerifierRouter(vm.envAddress("VERIFIER_ROUTER"));
+        console2.log("Using RiscZeroVerifierRouter at address", address(verifierRouter));
+
+        // Deploy new contracts
+        RiscZeroGroth16Verifier verifier =
```

Comment:
Note: I need `RiscZeroGroth16Verifier` here. I'd rather use `IRiscZeroVerifier`, but I need access to the verifier's `SELECTOR` field. Do we want to add the notion of "selector" to the interface?

Reply:
I didn't because it's not 1:1 in general. A router, for instance, has 0 or more selectors.